### ISSUE TITLE
Add google login 

### ISSUE THREAD

**Issue by prathaprathod** [thepracticaldev/dev.to#1595]
Hi , dev team 

I am using dev.to community and this website is superb and i need feature add google authentication people who doesn't havd twitter or github those people login through google or add email and password 

I hope this feature will consider

**Comment by jessleenyc** [comment]
Hey @prathaprathod, thanks for the feature request. We'd like to continue discussing this option as a team and community before approving the issue.

**Comment by rhymes** [comment]
I'll try to add my POV to the mix as a possible framework for the discussion.

The first reason, and the most obvious one, to add third party logins to a website is to facilitate the login process and lower the threshold of effort to the user. Increase user comfort. You are already a member of a reputable service and you give up a little bit of the data stored in such service to gain access to the website you're trying to become a member of.

This first aspect would be fully realised by adding any major or minor third party service, be it Google, Facebook, LinkedIn, Mastodon and many others. But this is just the most bare requirement which all of these services, in addition to Github and Twitter that are already available, have to met.

The second reason, the one I think it's the most interesting, is the community factor (for my lack of better words). Let's examine the two available options in this regard as an example:

* GitHub carries with it the banner of being the biggest community of open source developers. So by adding it to DEV, I presume, its founders tapped in the pre-existing community and the complimentary pride that sparks from showing the GitHub logo next to one's own profile on DEV

* Twitter, despite its limitations as a generic community, is still the de-facto network where that binds cross sections of developers and has islands of very active programmers that take pride in sharing their opinions, lifting one another and just be part of a network. I recently joined and I can follow all sorts of developers, especially people who do different things than I do everyday and learn stuff and pick up cultural nuances I could never have been exposed to before. Funnily enough, this is one of the reasons I spend time reading, writing, commenting and contributing to DEV in the first place. So, Twitter, is another service that (still my presumption) the co-founders decided to add because they thought that the exchange of "email and permission to read" was worth the being linked to such a big and transversal community.

Adding "Google login" would totally fall short of this second aspect. Being part of the Google federated network does not bring any value to the users, I'm not instantly part of a community of like minded people or people who do the same job that I do or share the same interests that I do. It would be like having an account on Amazon (another service one could consider as a third party login provider). The fact that we all have an Amazon account does not imply anything worth joining I think.

The third reason to is image. It's the one standing of less firm ground but it's still worth considering. Does DEV, by associating its gate locks to this or that service, add or substract value to its own image? 2018 has definitely NOT been the year of relationships between the common people (us) and the tech giants. 

So, if the social capital brought by adding Google Auth is nil and the image boost is dubious, the only remaining pull factor is making it a little easier for a (as in uknonwn yet) percentage of developers to log in. 

Is it worth it? Maybe. I don't know and it's not for me to decide. 

I hope this long thought of mine can be considered in the discussion.

🖖

**Comment by prathaprathod** [comment]
can u add goto top button

**Comment by maestromac** [comment]
@rhymes well said!

**Comment by FredoGonzalezR** [comment]
Well @rhymes has an interesting point.
But even if Google is not a worth option for authentication, I'd like to discuss and suggest to **add at least email login** and not depend on a third party to have access to dev.to.

What happens if Twitter or GitHub closes or I lose access to my account? They are communities too, and they could have their own rules different to dev.to's.
They are both tech giants like Google, and their support could be hard to reach out.

Or what if I just want to delete my account from that communities/services?

Something good about Google is that it is not a community oriented account, it is more unusual to get your account suspended or closed.

Currently I'm having issues with my Twitter account, ain't got response from their support so I'm closing it. I had to contact every service or community that I signed in with, and asked them to unlink my Twitter account, some didn't responded me so I'm losing access to those accounts too.

My point is that dev.to (and every website) should add another "non community" related sign in method, another way to keep our account safe and accessible for us.

🙋 +1 here for email sign in or Google sign in.
